### PR TITLE
0.3.0 release.

### PR DESCRIPTION
**Pull request by 12wrigja**

### CODE REVIEW

**Review by ptomato — APPROVED** [review]
:shipit:

**Review by justingrant — APPROVED** [review]
🎉  Should we add a changelog before release?

**Comment by 12wrigja** [comment]
>  Should we add a changelog before release?

Yep - I'll put one into this PR for you both to review.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:3]:
```suggestion
This version roughly corresponds with all the changes made to the Temporal polyfill as of the November 2021 TC39 Plenary meeting.
```

**Comment by 12wrigja** [comment]
I've drafted up a CHANGELOG and intend to reference it from the GH release notes. I looked over all 74 commits in this release and tried to summarize them, but there were definitely some normative changes that I wasn't sure of the external effects and left them out. If there's anything notable you think should be included, feel free to comment :)

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md]: I think commit IDs in MD files need to be HTTPS links, otherwise they'll show up as plain text to the reader.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:5]: Would "Breaking Changes" be clearer?

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:30]:
```suggestion
- Various `#round` and `#total` methods now accept string parameters or options bags. Strings are interpreted as the `smallestUnit` option (or `unit` for `Temporal. Duration#total`). (068e801f)
```

File: CHANGELOG.md
Comment (justingrant) [reply CHANGELOG.md]: BTW I'm not sure about this; in just not seeing the commits show up as links when editing here on my phone.

File: CHANGELOG.md
Comment (ptomato) [reply CHANGELOG.md]: There wasn't a November meeting - I think this should be October :smile:

File: package.json
Comment (justingrant) [inline package.json:4]:
```suggestion
  "description": "Polyfill for TC39 Stage 3 proposal Temporal (https://github.com/tc39/proposal-temporal)",
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md]:
```suggestion
 - The polyfill's source was ported to TypeScript (12e4d529, ac78fd9d, 53f32e0f, 06b806c9, 66fdc765, 50b1c34b, 4724b017, 947a8a5e, fdbf7e01, fa60af6a, da753f2f, f4db8b0b, 4a38420d)
```

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md]: Unfortunately you are right:  https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls notes that this doesn't happen for wikis or files within repos.

**Review by justingrant — APPROVED** [review]
Overall looks good. Only general suggestion is that it'd be good to link to the corresponding proposal-temporal PR when there is one.

**Review by ptomato — APPROVED** [review]
Looks generally good to me, I'll defer to Justin's more detailed comments.

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md]: I've added all the commit links, and added some notes in the release-checklist on how to quickly regenerate the list of links.

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md]: Yep!

File: package.json
Comment (12wrigja) [reply package.json]: Done.

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md]: Oops - Fixed :D

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:28]: Should we call this "Non-breaking changes" for symmetry, and move this section below the "Breaking changes" section?

Comment (12wrigja) [reply CHANGELOG.md:28]: Maybe call it "New features", but keep it where it is?

 Personally, I think "breaking changes > bug fixes > new stuff that isn't a bugfix > misc documentation for maintainers" is the right order.

Comment (justingrant) [reply CHANGELOG.md]: > Maybe call it "New features", but keep it where it is?
For a regular OSS package I'd agree, but as a Stage 3 polyfill claiming that there are new "features" seems like a bad idea because the idea of Stage 3 is that there won't be any new features!  I'd prefer to call them "Non-breaking changes" or some other wording that doesn't make it sound like the proposal is a moving target.

> Personally, I think "breaking changes > bug fixes > new stuff that isn't a bugfix > misc documentation for maintainers" is the right order.

Sure that's OK with me.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:34]: Is this really a non-breaking change?  Given that it introduces new exceptions for cases that didn't throw before, is it a bug fix or a breaking change?

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:44]: This is so cool. Thanks for doing this!

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md]: I suppose in terms of functionality there's no new features, though accepting strings to round and total sure seemed like a "feature" to me 😆 

Non-breaking definitely makes more sense with your explanation, so let's use that.

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md]: Agreed this is actually a bug fix, not a non-breaking change. Moved.

**Comment by 12wrigja** [comment]
Ok - one more PR merged in and added to the changelog.

@justingrant can I get one more look over before merging and starting a release? Thanks.

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md:45]: Of course! I hope to automate this process just a little more in the future so we can always be certain this changelog makes sense.

Something that would help greatly I think going forward would be some sort of check during PR review to either tag a PR with the relevant spec PRs, or to explicitly acknowledge that there are no corresponding PRs. I'm going to ponder this a bit over the holidays and might throw something together for us to consider in the new year.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:7]: Is there a corresponding proposal-temporal PR? OK if not, just wanted to make sure.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:21]: Minor nit: I don't think this was a spec change, so "Spec PR" isn't the right title for the link. Maybe "Related PR" ?  "Related proposal-temporal PR" ?  Or any other label you think would make sense.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:28]: Same here. Not a spec change.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:26]: Not a spec change.

**Review by justingrant — APPROVED** [review]
Overall looks great. A few notes about clarifying which links are actual spec changes and which are simply PRs ported from proposal-temporal or mirrored there.

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md:7]: There is a corresponding PR, but weirdly it doesn't change the spec text. Seems maybe the spec text was already  this way (for a while): https://github.com/tc39/proposal-temporal/blame/60c753a6369c0a61cacc353562214aaa6327117a/spec/timezone.html#L586-L597

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md]: Changed all the labels where the spec isn't changed to be "proposal-temporal polyfill PR".

**Pull request merged**
